src/natpmp.cpp

```diff
@@ -176,6 +176,21 @@ bool natpmp::should_log() const
 	return m_callback.should_log_portmap(aux::portmap_transport::natpmp);
 }
 
+void natpmp::mapping_log(char const* op, mapping_t const& m) const
+{
+	if (should_log())
+	{
+		log("%s-mapping: proto: %s port: %d local-port: %d action: %s ttl: %" PRId64
+			, op
+			, m.protocol == portmap_protocol::none
+			? "none" : to_string(m.protocol)
```

Comment:
it would probably make sense to bake-in support for "none" into ``to_string()`` and not check for it here. what do you think?